ansible_galaxy/models/collection_info.py

```diff
@@ -1,17 +1,79 @@
-
-import logging
+from __future__ import print_function
 
 import attr
+import json
+import logging
+import os
+import re
+import semver
 
-log = logging.getLogger(__name__)
+logger = logging.getLogger(__name__)
 
+TAG_REGEXP = re.compile('^[a-z0-9]+$')
 
 # see https://github.com/ansible/galaxy/issues/957
+
+
 @attr.s(frozen=True)
 class CollectionInfo(object):
-    namespace = attr.ib()
-    name = attr.ib()
-    version = attr.ib()
-    format_version = attr.ib(default=0.0)
-    author = attr.ib(default=None)
+    name = attr.ib(default=None)
+    version = attr.ib(default=None)
+    authors = attr.ib(default=[])
     license = attr.ib(default=None)
+    description = attr.ib(default=None)
+    keywords = attr.ib(default=[])
+    readme = attr.ib(default='README.md')
+    dependencies = attr.ib(default=[])
+
+    @property
+    def namespace(self):
+        return self.name.split('.', 1)[0]
+
+    @name.validator
+    @version.validator
+    @license.validator
+    @description.validator
+    def _check_required(self, attribute, value):
+        if value is None:
+            raise ValueError("Invalid collection metadata. '%s' is required" % attribute.name)
+
+    @version.validator
+    def _check_version_format(self, attribute, value):
+        try:
+            semver.parse_version_info(value)
+        except ValueError:
+            raise ValueError("Invalid collection metadata. Expecting 'version' to be in "
+                             "semantic version format.")
+
+    @license.validator
+    def _check_license(self, attribute, value):
+        cwd = os.path.dirname(os.path.abspath(__file__))
+        license_path = os.path.join(cwd, '..', 'data', 'spdx_licenses.json')
+        license_data = json.load(open(license_path, 'r'))
+        for license in license_data['licenses']:
+            if license['licenseId'] == value:
+                if license['isDeprecatedLicenseId']:
+                    print("Warning: collection metadata 'license' value %s is "
+                          "deprecated." % value)
+                return True
+        raise ValueError("Invalid collection metadata. Expecting 'license' to be a valid "
```

Comment:
Client probably shouldn't be enforcing a license type. The list of licences will need to be updated regularly.

Mazer should be able to build an artifact or install a collection of any sort of license. Maybe make it a optional 'oss_license' or 'spdx_license'?

Reply:
No. We need to enforce the license. We're either going to do SPDX, or we're not. Add the following the the release build process, please:

```
curl https://raw.githubusercontent.com/spdx/license-list-data/master/json/licenses.json --output spdx_licenses.json
```